### PR TITLE
CREATE TABLE check for duplicate columns

### PR DESCRIPTION
**Pull request by ebyhr**
I wrote the test in `TestAnalyzer.java` firstly, but the analyze does't cause exception.
Therefore, I wrote it in `AbstractTestIntegrationSmokeTest.java`.

### CODE REVIEW

**Comment by ebyhr** [comment]
#7846

File: presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java [hunk 1]
```diff
@@ -94,6 +95,9 @@ public String explain(CreateTable statement, List<Expression> parameters)
                 if ((type == null) || type.equals(UNKNOWN)) {
                     throw new SemanticException(TYPE_MISMATCH, column, "Unknown type for column '%s' ", column.getName());
                 }
+                if (columns.stream().anyMatch(c -> c.getName().equals(column.getName()))) {
+                    throw new SemanticException(DUPLICATE_COLUMN_NAME, column, "Column name '%s' specified more than once", column.getName());
+                }
                 columns.add(new ColumnMetadata(column.getName(), type, column.getComment().orElse(null), false));
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java]: The case of `element instanceof LikeClause` also adds columns. Shouldn't that add be covered with a check too?

File: presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java [hunk 2]
```diff
@@ -94,6 +95,9 @@ public String explain(CreateTable statement, List<Expression> parameters)
                 if ((type == null) || type.equals(UNKNOWN)) {
                     throw new SemanticException(TYPE_MISMATCH, column, "Unknown type for column '%s' ", column.getName());
                 }
+                if (columns.stream().anyMatch(c -> c.getName().equals(column.getName()))) {
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java]: I'm pretty sure this is not a big deal, but such O(n2) doesn't look good to me. What about turning columns into `LinkedHashMap<String /*name*/, ColumnMetadata>` ?

**Comment by ebyhr** [comment]
Updated.

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java [hunk 3]
```diff
@@ -135,6 +135,15 @@ public void testDescribeTable()
         assertTrue(expectedColumnsPossibilities.contains(actualColumns), String.format("%s not in %s", actualColumns, expectedColumnsPossibilities));
     }
 
+    @Test
+    public void testDuplicatedRowCreateTable()
+    {
+        assertQueryFails("CREATE TABLE test (a integer, a integer)",
```
Comment (martint) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java]: Add a test for columns that differ only in capitalization.

**Comment by ebyhr** [comment]
@martint Updated. I appended `toLowerCase()`.

**Pull request merged**